src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py

```diff
@@ -1537,3 +1541,19 @@ def test_data_scrubbing(self):
         with open(log_path, "r") as f:
             log_content = f.read()
         assert "**data_scrubbed**" in log_content
+
+    def test_cli_ua(self, pf):
+        # clear user agent before test
+        context = OperationContext().get_instance()
+        context.user_agent = ""
+        with environment_variable_overwrite(PF_USER_AGENT, ""):
+            with pytest.raises(SystemExit):
+                run_pf_command(
+                    "run",
+                    "show",
+                    "--name",
+                    "not_exist",
+                )
+        user_agent = setup_user_agent_to_operation_context()
+        ua_dict = parse_ua_to_dict(user_agent)
+        assert ua_dict.keys() == {"promptflow-sdk", "promptflow-cli", "promptflow"}
```

Comment:
why we have promptflow-sdk here?

Reply:
This command will go through SDK's logic, it refers to this log:  
![image](https://github.com/microsoft/promptflow/assets/7776147/88fd412d-f7d7-4172-9a42-0edff881d370)